streaming/plugins/abci/v1/README.md

```diff
@@ -0,0 +1,391 @@
+to the `ABCIListener` interface. One thing to note here is that plugin implementations
+defined in the `ListenerGRPCPlugin.Impl` property are **only** required when building
+plugins in Go. They are pre compiled into Go modules. The `GRPCServer.Impl` calls methods
+on this out-of-process plugin.
+
+For Go plugins this is all that is required to process data that is sent over gRPC.
+This provides the advantage of writing quick plugins that process data to different
+external systems (i.e: DB, File, DB, Kafka, etc.) without the need for implementing
+the gRPC server endpoints.
+
+```go
+// MyPlugin is the implementation of the baseapp.ABCIListener interface
+// For Go plugins this is all that is required to process data sent over gRPC.
+type MyPlugin struct {
+	...
+}
+
+func (a FilePlugin) ListenBeginBlock(ctx context.Context, req abci.RequestBeginBlock, res abci.ResponseBeginBlock) error {
+	// process data
+	return nil
+}
+
+func (a FilePlugin) ListenEndBlock(ctx context.Context, req abci.RequestEndBlock, res abci.ResponseEndBlock) error {
+    // process data
+    return nil
+}
+
+func (a FilePlugin) ListenDeliverTx(ctx context.Context, req abci.RequestDeliverTx, res abci.ResponseDeliverTx) error {
+    // process data
+    return nil
+}
+
+func (a FilePlugin) ListenCommit(ctx context.Context, res abci.ResponseCommit, changeSet []*store.StoreKVPair) error {
+    // process data
+    return nil
+}
+
+func main() {
+	plugin.Serve(&plugin.ServeConfig{
+		HandshakeConfig: v1.Handshake,
+		Plugins: map[string]plugin.Plugin{
+			"abci_v1": &v1.ABCIListenerGRPCPlugin{Impl: &MyPlugin{}},
+		},
+
+		// A non-nil value here enables gRPC serving for this streaming...
+		GRPCServer: plugin.DefaultGRPCServer,
+	})
+}
+```
+
+## Plugin Loading System
+
+A general purpose plugin loading system has been provided by the SDK to be able to load not just
+the `ABCIListener` service plugin but other protocol services as well. You can take a look
+at how plugins are loaded by the SDK in [streaming/streaming.go](https://github.com/cosmos/cosmos-sdk/blob/main/streaming/streaming.go)
+
+You'll need to add this in your `app.go`
+```go
+// app.go
+
+func NewApp(...) *App {
+
+    ...
+
+    // register streaming services
+    streamingCfg := cast.ToStringMap(appOpts.Get(baseapp.StreamingTomlKey))
+    for service := range streamingCfg {
+        pluginKey := fmt.Sprintf("%s.%s.%s", baseapp.StreamingTomlKey, service, baseapp.StreamingABCIPluginTomlKey)
+        pluginName := strings.TrimSpace(cast.ToString(appOpts.Get(pluginKey)))
+        if len(pluginName) > 0 {
+            logLevel := cast.ToString(appOpts.Get(flags.FlagLogLevel))
+            plugin, err := streaming.NewStreamingPlugin(pluginName, logLevel)
+            if err != nil {
+                tmos.Exit(err.Error())
+            }
+            if err := baseapp.RegisterStreamingPlugin(bApp, appOpts, keys, plugin); err != nil {
+                tmos.Exit(err.Error())
+            }
+        }
+    }
```

Comment:
Is there a reason to not expose this as a function for consuming applications to use, e.g. `RegisterStreamingServices`?

Reply:
No reason. We had a discussion about moving that into the `BaseApp` so app builders would not forget to implement it or have to call it at all. But it was something that the SDK team wanted to handle post merge. I haven't circled back to it yet.

(ref: https://github.com/cosmos/cosmos-sdk/pull/13473#discussion_r990118166)